addon/mixins/authenticated-route-mixin.js

```diff
@@ -61,6 +61,10 @@ export default Mixin.create({
     @public
   */
   session: service('session'),
+  router: computed(function() {
+    return this.get('routerService') ? this.get('routerService') : getOwner(this).lookup('router:main');
+  }),
+  routerService: service('router'),
```

Comment:
This will raise if the router service does not exist which is the case for older versions of Ember. I think what you'll want to do instead is:

```js
router: computed(function() {
  let owner = getOwner(this);
  return owner.lookup('service:router') || owner.lookup('router:main');
}),
```

which will use the router service if it exists and fall back to the main router if it doesn't.

Reply:
Hi @marcoow or @mike-north can one of you check that this can be merged now  that we have a robust  call for the router?